front/src/reducers/osrdconf/operationalStudiesConf/powerRestrictionReducer.ts

```diff
@@ -163,42 +163,31 @@ const powerRestrictionReducer = {
     action: PayloadAction<{
       firstRestriction?: PowerRestriction;
       secondRestriction: PowerRestriction;
+      endPosition: number;
       newFromPathStep: PathStep;
```

Comment:
To avoid having to do `newFromPathStep.positionOnPath!` l.178, we could use the custom type `NonNullableObject` (after exporting it in src/utils/types) like this : 
```suggestion
      newFromPathStep: NonNullableObject<PathStep, 'positionOnPath'>;
```
This way `positionOnPath` will be define and we don't have to deal with it later. 
This implies to change `PathStep` type in a few other functions, I let you decide if you find it worth it or not :)

Reply:
Yes I agree. Can we do this when we merge the 2 actions ?

Reply:
Sure